### ISSUE TITLE
Incorrect value of kind attribute in spring-batch-3.0.xsd

### ISSUE THREAD

**Issue by spring-projects-issues** [spring-projects/spring-batch#1479]
**[Harold Thor Eyrich](https://jira.spring.io/secure/ViewProfile.jspa?name=thunderheadthor)** opened **[BATCH-2117](https://jira.spring.io/browse/BATCH-2117?redirect=false)** and commented

The value "parent" is not valid for attribute kind on the annotation for element flow in spring-batch-2.2.xsd.  Attribute definition from spring-tool-3.1.xsd:
<xsd:attribute name="kind" default="direct">
<xsd:simpleType>
<xsd:restriction base="xsd:string">
<xsd:enumeration value="ref"/>
<xsd:enumeration value="direct"/>
</xsd:restriction>
</xsd:simpleType>
</xsd:attribute>

---

**Attachments:**
- [xsdError.PNG](https://jira.spring.io/secure/attachment/21424/xsdError.PNG) (_61.65 kB_)

**Comment by benas** [comment]
Spring Batch v2 and its xsd are not supported anymore, but this is still a valid issue in `spring-batch-3.0.xsd` which is used by the latest v4.3. The fix consists in changing the value of the `kind` attribute in [Line 482](https://github.com/spring-projects/spring-batch/blob/master/spring-batch-core/src/main/resources/org/springframework/batch/core/configuration/xml/spring-batch-3.0.xsd#L482) as follows:

```diff
-- <tool:annotation kind="parent">
++ <tool:annotation kind="ref">
```

**Comment by bbohosyan** [comment]
Can I take it ? I want to start contributing and this seems like a perfect first issue to solve and get used to.

**Comment by benas** [comment]
@bbohosyan Sure, you are welcome! There are no tests to add/update for this change. You can open a PR and I will link it to this issue.

**Comment by benas** [comment]
Fixed in #3844 .
